src/Validator.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * This file is part of Lcobucci\JWT, a simple library to handle JWT and JWS
+ *
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD-3-Clause
+ */
+
+declare(strict_types=1);
+
+namespace Lcobucci\JWT;
+
+use Lcobucci\JWT\Validation\Constraint;
+use Lcobucci\JWT\Validation\InvalidTokenException;
+
+/**
+ * @author Luís Otávio Cobucci Oblonczyk <lcobucci@gmail.com>
+ * @author Danny Dörfel <danny.dorfel@gmail.com>
+ * @author Marco Pivetta <ocramius@gmail.com>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ *
+ * @since 4.0.0
+ */
+interface Validator
```

Comment:
> Sorry, I don't know if this was already discussed, but how do we want to use Validators?

```php
$config = new Configuration();
$validator = $config->getValidator();

$validator->validate(
     $token,
     new ValidAt(new DateTimeImmutable()),
     new IssuedBy('test.com')
);

// or simply
$constraints = [new ValidAt(new DateTimeImmutable()), new IssuedBy('test.com')];
$validator->validate($token, ...$constraints);
```

> As Configuration serves like a small DI container, there are also other possibilities, e.g.:

I don't see changing the state of the service as an option, IMO that's not a good practice. Also, `new ValidAt(new DateTimeImmutable())` might lead to some issues due to the creation and the validation time.